nexus/tests/integration_tests/subnet_allocation.rs

```diff
@@ -72,53 +84,91 @@ async fn test_subnet_allocation(cptestctx: &ControlPlaneTestContext) {
         organization_name, project_name
     );
 
-    // Modify the default VPC to have a very small subnet so we don't need to
-    // issue many requests
-    let url_subnet = format!(
-        "/organizations/{}/projects/{}/vpcs/default/subnets/default",
+    // Create a new, small VPC Subnet, so we don't need to issue many requests
+    // to test address exhaustion.
+    let url_subnets = format!(
+        "/organizations/{}/projects/{}/vpcs/default/subnets",
         organization_name, project_name
     );
-    let subnet = "192.168.42.0/29".parse().unwrap();
-    let subnet_update = params::VpcSubnetUpdate {
-        identity: IdentityMetadataUpdateParams {
-            name: Some("default".parse().unwrap()),
-            description: None,
+    let subnet_name = "small";
+    let subnet = "192.168.42.0/26".parse().unwrap();
+    let subnet_create = params::VpcSubnetCreate {
+        identity: IdentityMetadataCreateParams {
+            name: subnet_name.parse().unwrap(),
+            description: String::from("a small subnet"),
         },
-        ipv4_block: Some(Ipv4Net(subnet)),
+        // Use the minimum subnet size
+        ipv4_block: Ipv4Net(subnet),
         ipv6_block: None,
     };
-    NexusRequest::object_put(client, &url_subnet, Some(&subnet_update))
+    NexusRequest::objects_post(client, &url_subnets, &Some(&subnet_create))
         .authn_as(AuthnMode::PrivilegedUser)
         .execute()
         .await
         .unwrap();
 
     // The valid addresses for allocation in `subnet` are 192.168.42.5 and
     // 192.168.42.6. The rest are reserved as described in RFD21.
-    create_instance(client, organization_name, project_name, "i1").await;
-    create_instance(client, organization_name, project_name, "i2").await;
+    let nic = params::InstanceNetworkInterfaceAttachment::Create(vec![
+        params::NetworkInterfaceCreate {
+            identity: IdentityMetadataCreateParams {
+                name: "eth0".parse().unwrap(),
+                description: String::from("some iface"),
+            },
+            vpc_name: "default".parse().unwrap(),
+            subnet_name: "small".parse().unwrap(),
+            ip: None,
+        },
+    ]);
+
+    // Create a shit-ton of instances, up to the size of the IP subnet.
```

Comment:
Just to clarify, how many instances are we creating here?

Just wanna validate this test isn't taking an excessively long time to execute

Reply:
There should be 58 instances created. Subnet is a `/26`, so total size is `2 ** 6 - 6 = 64 - 6 = 58`. "Shit-ton" is the technical term :)

Reply:
To clarify, a `/26` is the smallest subnet the API will allow users to create.

Reply:
I did some rough timing. The test is definitely much longer. I don't see a way to exercise the failure modes we're looking for without it, though, since we're explicitly testing exhaustion of the IP addresses within a subnet.

With this change:

```
bnaecker@feldspar : ~/omicron $ time cargo t -- test_subnet_allocation
....
real	1m16.451s
user	1m13.553s
sys	0m9.746s
```

Before this change:

```
bnaecker@feldspar : ~/omicron $ time cargo t -- test_subnet_allocation
....
real	0m25.889s
user	0m17.551s
sys	0m6.602s
```

In both cases, that includes a lot of time just filtering out the tests we don't run. This doesn't seem overly long to me, but I'm definitely open to suggestions. That could include just deleting the test BTW. It's really a test of the public error message in this case, since there are already a number of tests verifying the DB query fails in the right when we the IP subnet is exhausted.

Reply:
Merging at this point, but we might revisit this if the duration becomes a problem.